include/transformations/RenameEntitiesTransformation.h

```diff
@@ -65,44 +61,28 @@ class RenameEntitiesVisitor : public RecursiveASTVisitor<RenameEntitiesVisitor>
     const bool rename_func = false;
     const bool rename_var = false;
 
-    discrete_distribution<int> token_len_generator;
-    discrete_distribution<int> tokens_num_generator;
-    discrete_distribution<int> char_generator;
-
-    mt19937 * gen;
-
-    const bool test = false;
-
-    const bool with_hash = false;
-    const string hash_prefix = "d";
+    BaseRenameProcessor * processor;
 
     map<Decl *, string> decl2name;
-    string randomSnakeCaseName();
-    string hashName(string * name);
     void processVarDecl(Decl * decl, string * name);
 };
 
 class RenameEntitiesASTConsumer : public ASTConsumer {
  public:
     explicit RenameEntitiesASTConsumer(Rewriter * rewriter, const bool rename_func, const bool rename_var,
-                                       discrete_distribution<int> token_len_generator,
-                                       discrete_distribution<int> tokens_num_generator,
-                                       discrete_distribution<int> char_generator,
-                                       mt19937 * gen, const string * hash_prefix,
-                                       const bool with_hash, const bool test);
+                                       BaseRenameProcessor * processor);
     void HandleTranslationUnit(ASTContext &ctx); // NOLINT.
  private:
     RenameEntitiesVisitor visitor;
     Rewriter * rewriter;
+    BaseRenameProcessor * processor;
```

Comment:
Is it necessary to have `BaseRenameProcessor ` as an attribute of `RenameEntitiesASTConsumer`?